### PR TITLE
writing HI configuration file 2024

### PR DESCRIPTION
**Pull request by LinaresToine**
HI datataking is coming up and I attempt to create the HIReplayConfiguration.py file. 

Questions remaining: 
* Global tags
* Scenarios
* HLTScout testing configuration
* PDs configuration

### CODE REVIEW

**Comment by germanfgv** [comment]
It seems you are replacing HI replay config with the regular replay config. Please review the changes, as there are many parameters that are different from HI to pp configs.

**Comment by LinaresToine** [comment]
Yes, I originally created the file based on current replay config. I now based it on last year's HI config. 

I believe the file is ready to run a replay, the only thing we miss for now are 141X GTs.

File: etc/HIReplayOfflineConfiguration.py [hunk 1]
```diff
@@ -136,8 +149,8 @@
 alcarawProcVersion = dt
 
 # Defaults for GlobalTag
-expressGlobalTag = "132X_dataRun3_Express_v4"
-promptrecoGlobalTag = "132X_dataRun3_Prompt_v4"
+expressGlobalTag = "141X_dataRun3_Express_v2"
+promptrecoGlobalTag = "141X_dataRun3_Prompt_v2"
```
Comment (mmusich) [inline etc/HIReplayOfflineConfiguration.py:153]:
```suggestion
promptrecoGlobalTag = "141X_dataRun3_Prompt_v3"
```
`141X_dataRun3_Prompt_v3` has `DeDxCalibrationRcd`, see https://github.com/cms-sw/cmssw/issues/46186

File: etc/HIReplayOfflineConfiguration.py [hunk 2]
```diff
@@ -1337,7 +1373,7 @@
                write_dqm=True,
                alca_producers=["TkAlMinBias"],
                dqm_sequences=["@common"],
-               scenario=hiTestppScenario)
+               scenario=hiLightEventsScenario)
```
Comment (mmusich) [inline etc/HIReplayOfflineConfiguration.py:1376]: @LinaresToine I don't understand this choice (and also several others).
In 2023 this PD was processed with `ppEra_Run3_pp_on_PbPb_2023`. I think this should be `hiHeavyEventsScenario` (= `"ppEra_Run3_pp_on_PbPb_2024"`) now.

File: etc/HIReplayOfflineConfiguration.py [hunk 3]
```diff
@@ -1346,7 +1382,7 @@
                do_reco=False,
                aod_to_disk=False,
                raw_to_disk=False,
-               scenario=hiTestppScenario)
+               scenario=hiLightEventsScenario)
```
Comment (mmusich) [inline etc/HIReplayOfflineConfiguration.py:1385]: @LinaresToine I don't understand this choice (and also several others).
In 2023 this PD was processed with `ppEra_Run3_pp_on_PbPb_2023`. I think this should be `hiHeavyEventsScenario` (= `"ppEra_Run3_pp_on_PbPb_2024"`) now.

File: etc/HIReplayOfflineConfiguration.py [hunk 4]
```diff
@@ -1357,7 +1393,7 @@
                raw_to_disk=False,
                aod_to_disk=False,
                dqm_sequences=["@common"],
-               scenario=hiTestppScenario)
+               scenario=hiLightEventsScenario)
```
Comment (mmusich) [inline etc/HIReplayOfflineConfiguration.py:1396]: @LinaresToine I don't understand this choice (and also several others).
In 2023 this PD was processed with `ppEra_Run3_pp_on_PbPb_2023`. I think this should be `hiHeavyEventsScenario` (= `"ppEra_Run3_pp_on_PbPb_2024"`) now.

File: etc/HIReplayOfflineConfiguration.py [hunk 5]
```diff
@@ -1433,7 +1469,7 @@
                timePerEvent=1,
                disk_node="T2_US_Vanderbilt",
                dqm_sequences=["@commonSiStripZeroBias"],
-               scenario=hiTestppScenario)
+               scenario=hiLightEventsScenario)
```
Comment (mmusich) [inline etc/HIReplayOfflineConfiguration.py:1472]: @LinaresToine I don't understand this choice (and also several others).
In 2023 this PD was processed with `ppEra_Run3_pp_on_PbPb_2023`. I think this should be `hiHeavyEventsScenario` (= `"ppEra_Run3_pp_on_PbPb_2024"`) now.

**Comment by LinaresToine** [comment]
Thank you @mmusich . I originally understood that RawPrime goes with `ppEra_Run3_pp_on_PbPb_approxSiStripClusters_2024`, while everything else goes with UPC, except for express, which use `ppEra_Run3_pp_on_PbPb_2024`. I'll retry with:

RawPrime : `ppEra_Run3_pp_on_PbPb_approxSiStripClusters_2024` --> hiRawPrimeScenario
Forward: `ppEra_Run3_2024_UPC` --> hiForwardScenario
Everything else: `ppEra_Run3_pp_on_PbPb_2024` --> hiScenario

**Comment by fabiocos** [comment]
@LinaresToine we should try a new replay test of this configuration using CMSSW_14_1_1, that should contain all the latest CMSSW updates discussed as a follow-up of the previous failure, in preparation for the forthcoming HI data taking (regular pp data taking is ending in one week from now)

**Comment by LinaresToine** [comment]
For the record, previous issues with cmsssw were solved in this issue:

* https://github.com/cms-sw/cmssw/issues/46186
   * Includes solution to problem with UPC scenario
   * Includes solution related to outdated global tags

* https://github.com/cms-sw/cmssw/issues/46307
   * Reports current problem with DeDxEstimator module in HIForward job

**Comment by fabiocos** [comment]
https://github.com/cms-sw/cmssw/releases/tag/CMSSW_14_1_2 is now available with the known needed fixes. I suggest to update the branch above, and restart a replay before the week-end, so as we may have at the Joint Ops on Monday a clear view of the status in preparation of HI data taking

**Comment by germanfgv** [comment]
> https://github.com/cms-sw/cmssw/releases/tag/CMSSW_14_1_2 is now available with the known needed fixes. I suggest to update the branch above, and restart a replay before the week-end, so as we may have at the Joint Ops on Monday a clear view of the status in preparation of HI data taking

We will trigger the replay again, now with this new version. It should be done by Monday.

**Comment by LinaresToine** [comment]
The replay has started, it can be followed in 

[CMSSW_14_1_2 Replay](https://monit-grafana.cern.ch/d/t_jr45h7k/cms-tier0-replayid-monitoring?orgId=11&refresh=1m&var-Bin=5m&var-ReplayID=241011175940&var-JobType=All&var-WorkflowType=All)

**Comment by fabiocos** [comment]
The replay is successfully finished, both according to Grafana dashboard and the testbed WMStats instance

**Comment by LinaresToine** [comment]
Output data for replay can be found in https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=dataset%3D%2F*%2F*v11175940*%2F*

**Pull request merged**